crypto/http/http_lib.c

```diff
@@ -54,6 +61,7 @@ int OSSL_parse_url(const char *url, char **pscheme, char **puser, char **phost,
     init_pstring(puser);
     init_pstring(phost);
     init_pstring(pport);
+    init_pint(pport_num);
```

Comment:
I really do not understand how this change can help. Without it, the `*pport_num` will be left uninitialized only in case the OSSL_parse_url() call fails. Why that could change anything? IMO this is not the cause for the problem you've seen @rsbeckerca. I suspect there is something more subtle with these tests in regards to timing, etc.

Reply:
The line:
https://github.com/openssl/openssl/blob/168d93a21d512028572777ea5bc96994f2df6c36/crypto/http/http_lib.c#L113
does not use `sscanf` correctly. Specifying `!scanf` strictly checks for 0 vs. non-zero instead of what would be semantically better as `sscanf(...) != 1`, since `sscanf` returns the number of arguments processed rather than an error. I understand that that check is probably fine. Also, there is a redundant initialization of `pport` in `OSSL_HTTP_parse_url`.

I am just trying to be safe about initialized values on the stack. If the PR does not matter, it can be dropped.

Reply:
Good point that the result of `sscanf()` should better be checked using `!= 1`.
Yet in this case with one variable to be filled, using `== 0` boils down to the same.

Reply:
The call `init_pstring(pport);` in `OSSL_HTTP_parse_url()` is not redundant -
note that the subsequent
```
    if (!OSSL_parse_url(url, &scheme, puser, phost, &port, pport_num,
                        ppath, pquery, pfrag))
        return 0;
```
uses a different port variable pointer (`&port`) and may return on error, and the initialization is for that case.